calico_node/startup/startup.go

```diff
@@ -817,6 +822,31 @@ func ensureFelixConfig(c *client.Client, host, key, def string) error {
 	}
 }
 
+// ensureClusterTypeSet ensures that the ClusterType in the Felix global
+// config has the data contained in the passed in clusterType.  This includes
+// merging what was already set in ClusterType with the passed in clusterType.
+func ensureClusterTypeSet(c *client.Client, key, clusterType string) error {
+	global := ""
+	if val, assigned, err := c.Config().GetFelixConfig("ClusterType", global); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetFelixConfig(key, global, clusterType)
+	} else {
+		curCt := strings.Split(val, ",")
+	NEXT:
```

Comment:
That code only has to worry about one value and doesn't have to tokenize both strings and comparing each element.
I can remove the NEXT anchor by adding a flag that is set if the cluster type is found, would that be preferable?